components/brave_wallet/browser/internal/hd_key_ed25519_unittest.cc

```diff
@@ -250,56 +235,40 @@ TEST(HDKeyEd25519UnitTest, SignAndVerify) {
   std::vector<uint8_t> bytes;
   EXPECT_TRUE(
       base::HexStringToBytes("000102030405060708090a0b0c0d0e0f", &bytes));
-  auto key = HDKeyEd25519::GenerateFromSeed(bytes);
+  auto key = HDKeyEd25519::GenerateFromSeedAndPath(bytes, "m");
   const std::vector<uint8_t> msg_a(32, 0x00);
   const std::vector<uint8_t> msg_b(32, 0x08);
-  const std::vector<uint8_t> sig_a = key->Sign(msg_a);
-  const std::vector<uint8_t> sig_b = key->Sign(msg_b);
+  const auto sig_a = key->Sign(msg_a);
+  const auto sig_b = key->Sign(msg_b);
 
-  EXPECT_TRUE(key->VerifyForTesting(msg_a, sig_a));
-  EXPECT_TRUE(key->VerifyForTesting(msg_b, sig_b));
+  EXPECT_TRUE(VerifySignature(*key, msg_a, sig_a));
+  EXPECT_TRUE(VerifySignature(*key, msg_b, sig_b));
 
   // wrong signature
-  EXPECT_FALSE(key->VerifyForTesting(msg_a, sig_b));
-  EXPECT_FALSE(key->VerifyForTesting(msg_b, sig_a));
-
-  // signature size != 64
-  EXPECT_FALSE(key->VerifyForTesting(msg_a, std::vector<uint8_t>(128, 0xff)));
-  EXPECT_FALSE(key->VerifyForTesting(msg_a, std::vector<uint8_t>(32, 0xff)));
-  EXPECT_FALSE(key->VerifyForTesting(msg_b, std::vector<uint8_t>(128, 0xff)));
-  EXPECT_FALSE(key->VerifyForTesting(msg_b, std::vector<uint8_t>(32, 0xff)));
+  EXPECT_FALSE(VerifySignature(*key, msg_a, sig_b));
+  EXPECT_FALSE(VerifySignature(*key, msg_b, sig_a));
 }
 
 TEST(HDKeyEd25519UnitTest, GenerateFromPrivateKey) {
-  std::vector<uint8_t> private_key;
-  ASSERT_TRUE(base::HexStringToBytes(
-      "2b4be7f19ee27bbf30c667b642d5f4aa69fd169872f8fc3059c08ebae2eb19e7",
-      &private_key));
-  auto master_key = HDKeyEd25519::GenerateFromPrivateKey(private_key);
+  std::array<uint8_t, 64> key_pair;
+  ASSERT_TRUE(base::HexStringToSpan(
+      "2b4be7f19ee27bbf30c667b642d5f4aa69fd169872f8fc3059c08ebae2eb19e7"
+      "a4b2856bfec510abab89753fac1ac0e1112364e7d250545963f135f2a33188ed",
+      base::span(key_pair)));
+  auto master_key = HDKeyEd25519::GenerateFromKeyPair(key_pair);
   EXPECT_EQ(master_key->GetBase58EncodedKeypair(),
             "sCzwsBKmKtk5Hgb4YUJAduQ5nmJq4GTyzCXhrKonAGaexa83MgSZuTSMS6TSZTndnC"
             "YbQtaJQKLXET9jVjepWXe");
   EXPECT_EQ(master_key->GetBase58EncodedPublicKey(),
             "C5ukMV73nk32h52MjxtnZXTrrr7rupD9CTDDRnYYDRYQ");
-  EXPECT_EQ(master_key->GetPath(), "");
-
-  // 31 bytes
-  private_key.clear();
-  ASSERT_TRUE(base::HexStringToBytes(
-      "2b4be7f19ee27bbf30c667b642d5f4aa69fd169872f8fc3059c08ebae2eb19",
-      &private_key));
-  EXPECT_FALSE(HDKeyEd25519::GenerateFromPrivateKey(private_key));
 
-  // 33 bytes
-  private_key.clear();
-  ASSERT_TRUE(base::HexStringToBytes(
-      "2b4be7f19ee27bbf30c667b642d5f4aa69fd169872f8fc3059c08ebae2eb19e7ff",
-      &private_key));
-  EXPECT_FALSE(HDKeyEd25519::GenerateFromPrivateKey(private_key));
+  EXPECT_EQ(
```

Comment:
Is this changing because it's a keypair now we're expecting rather than just the private key? That's what I'm understanding in the code, but wanted to confirm.

Reply:
Yes. Rust implementation did pubkey derivation from private key. OpenSSl doesn't expose such derivation so we are importing both parts and validate them match each other